pos-print/src/test/java/com/clouway/pos/print/core/BackgroundReceiptPrintingServiceTest.kt

```diff
@@ -0,0 +1,155 @@
+package com.clouway.pos.print.core
+
+import com.clouway.pos.print.persistent.DatastoreCleaner
+import com.clouway.pos.print.persistent.DatastoreRule
+import com.clouway.pos.print.printer.Status
+import org.jmock.AbstractExpectations.returnValue
+import org.jmock.AbstractExpectations.throwException
+import org.jmock.Expectations
+import org.jmock.Mockery
+import org.jmock.integration.junit4.JUnitRuleMockery
+import org.jmock.lib.concurrent.Synchroniser
+import org.junit.ClassRule
+import org.junit.Rule
+import org.junit.Test
+import java.io.IOException
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+
+/**
+ * @author Tsvetozar Bonev (tsvetozar.bonev@clouway.com)
+ */
+class BackgroundReceiptPrintingServiceTest {
+
+  @Rule
+  @JvmField
+  var context = JUnitRuleMockery()
+
+  init {
+      context.setThreadingPolicy(Synchroniser())
+  }
+
+  companion object {
+    @ClassRule
+    @JvmField
+    val dataStoreRule = DatastoreRule()
```

Comment:
If you use ```@Rule``` instead of ```@ClassRule``` there is no need to make ```dataStoreRule``` static

Reply:
Making the annotation @Rule throws a null pointer and the only way I've managed to get this rule to work is by making it a static ClassRule.